File: app/src/main/java/com/example/jianming/Utils/NetworkUtil.java
package com.example.jianming.Utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import com.squareup.okhttp.OkHttpClient;

public class NetworkUtil {

    public static boolean isNetworkAvailable(Context context) {
        ConnectivityManager connectivityManager = ((ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE));
        if (connectivityManager != null) {
            NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
            return networkInfo != null && networkInfo.isConnected();
        } else {
            return false;
        }
    }

    public static OkHttpClient getOkHttpClient() {
        return okHttpClient;
    }

    private static OkHttpClient okHttpClient = new OkHttpClient();
}

File: app/src/main/java/com/example/jianming/Utils/NetworkUtil.kt
package com.example.jianming.Utils

import android.content.Context
import android.net.ConnectivityManager
import okhttp3.OkHttpClient

object NetworkUtil {
    fun isNetworkAvailable(context: Context): Boolean {
        val connectivityManager = context.getSystemService(ConnectivityManager::class.java)
        val currentNetWork = connectivityManager.activeNetwork
        return currentNetWork != null
    }

    val okHttpClient = OkHttpClient()
}